### ISSUE TITLE
hacker_news_pwa: Publish on Firebase/Github.io

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/angular#939]
Just fun :)

**Comment by matanlurey** [comment]
I'd like to set this up via travis, but we aren't green right now.

For now I'll do a single manual push.

**Comment by matanlurey** [comment]
Done as https://dart-lang.github.io/angular/ for now.

... can find a better home later on :)

**Comment by filiph** [comment]
That's pretty awesome.

![screen shot 2018-02-26 at 19 50 15](https://user-images.githubusercontent.com/919717/36710104-c249195a-1b30-11e8-88d3-1920a79407b0.png)

The page loads & paints & gets interactive in 2780ms, even on the simulated slow connection. That's really good for something like AngularDart, which doesn't even try to optimize for microbenchmarks like this. Just look at https://hnpwa.com/, the latest React score is 91 and TTI is 2570ms, AngularJS gets 4-6 seconds, etc.

Ignoring SEO (that's another discussion altogether), there are only a couple of issues. 

1. We lose some perf points by not caching `angular/main.dart.js` and `images/logo.svg` for long enough. I don't think there's anything we can do on github.io, but on Firebase Hosting, you just increase TTL to something like 24 hours and Lighthouse is happy (because that's when ISPs start to cache things, which makes things much faster for the end user). So: impossible fix on github.io, easy fix on firebase hosting.

2. The rest of points lost is "perceptual speed index". This is best explained [here](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/metrics/speed-index). Basically, the earlier we can show the UI (even if it's only a placeholder) the better. Maybe show the top bar in the static HTML? Even better, "make room" for the loaded articles with white divs? That alone makes a lot of the initial pixels correspond to the final result.

    And this is not just about "beating a score", by the way. The apps that do this kind of optimization _do_ perceptually load faster. It's a very different experience looking at a gray screen for 2 seconds than at the beginnings of a UI.

Congrats to all!

In summary, consider:

* Moving this to Firebase Hosting and cache the two files more aggressively.
* [Submitting](https://github.com/tastejs/hacker-news-pwas/blob/master/CONTRIBUTING.md) the app to https://hnpwa.com/.
* Adding the most visually prominent parts of the UI (app bar, white area with list of articles) to the static HTML + CSS.

**Comment by matanlurey** [comment]
Yeah, @leonsenft had the original version on Firebase Hosting. This is definitely temporary; I'd like to set it up to auto-push to Firebase Hosting every successful merge:

* [ ] Move to Firebase Hosting
* [ ] Submit to hnpwa.com
* [ ] Add visually prominent parts of the UI to the static HTML/CSS if possible

**Comment by leonsenft** [comment]
We already have it hosted on Firebase: https://dart-hn.firebaseapp.com/, although this is an older version. Now that it's open-sourced, it should be easier to keep in sync.

Are we happy with this location?

**Comment by matanlurey** [comment]
@leonsenft Seems fine to me unless @kevmoo has other ideas.

**Comment by kevmoo** [comment]
Let's run with it!

**Comment by matanlurey** [comment]
I have push access, so I'll try and setup an auto-push this week.

**Comment by matanlurey** [comment]
(We need our travis to be green, ideally)

**Comment by matanlurey** [comment]
<img width="668" alt="screen shot 2018-02-27 at 11 03 32 am" src="https://user-images.githubusercontent.com/168174/36748824-e27dda26-1bad-11e8-9985-fd6ececec949.png">

Similar numbers with Firebase @filiph, but thanks for suggesting that!

**Comment by filiph** [comment]
I think you need to bump up the `max-age` to something like 86400 for this to register. I can help with that, just point me to your `firebase.json`.

Note that yes, updates to `main.dart.js` will take up to 24 hours (86400 seconds) to show up without force refresh. In a production app, you'd have a build step that does things like `main.949234.dart.js` and then caches forever.

**Comment by leonsenft** [comment]
I agree it's pretty low priority, but perhaps we could automate bumping the number (in main.#.dart.js) in the Travis deploy script? Of course we'd need to do this for each script (including all of the deferred parts) as well as the list of offline URLs used by the service worker itself.

**Comment by kevmoo** [comment]
@filiph – if you have specific ideas/feedback, please file issues and add to this milestone: https://github.com/dart-lang/angular/milestone/7

**Comment by filiph** [comment]
@kevmoo I added https://github.com/dart-lang/angular/issues/963 (can't add it to milestone because I don't have rights here — should I be offended?).

**Comment by matanlurey** [comment]
I think @filiph's suggestion of just a ~10 minute cache is fine here.

**Comment by leonsenft** [comment]
Didn't @filiph say it should be 24 hours though to improve caching through ISPs?

**Comment by matanlurey** [comment]
I'm closing this issue for now, since we have something.

Please open additional issues for tracking.

**Comment by filiph** [comment]
Yeah, I meant longer than 10 minutes. 24 hours seems to be the customary max-age for js/image assets.
